### PR TITLE
#59 Product slider ADDED

### PR DESCRIPTION
**Pull request by Sky-De**
## What did i do ?

1. I've decided connect to server with my own mongoDb url but as there was no data or images  i just created mock data with images in code and implemented Carousel ( responsive ) using that, now it is commented for test cases , if you want you can remove that

2. I've added styles to index.css to customizing react-responsive-carousel  

## Show case gif of final result - It takes a few second to load gifs

**Desktop**
sliderShowCaseD-min

**Mobile**
sliderShowCaseM-min

if you got any question feel free to ask, otherwise merge SECOND COMMIT
thanks

### CODE REVIEW

**Comment by Sky-De** [comment]
I don't have access to take a look why deployment has failed, please provide screen shot of corresponding deployment error or if you can just fix it by yourself, let me know 
Thanks

**Comment by victorchrollo14** [comment]
> 1. I've decided connect to server with my own mongoDb url but as there was no data or images  i just created mock data with images in code and implemented Carousel ( responsive ) using that, now it is commented for test cases , if you want you can remove that

I have given it in the readme, you should run this to insert data into mongodb
if you have specified the mongodb uri in .env file, it would work fine.
``` 
cd server/src/utils/

node insertData.js
```

**Comment by victorchrollo14** [comment]
> I don't have access to take a look why deployment has failed, please provide screen shot of corresponding deployment error or if you can just fix it by yourself, let me know Thanks

seems like you forgot to add package.json file to staging before making your commit as a result it was not installed.

image

**Comment by victorchrollo14** [comment]
* Change those next and previous buttons to something similar to the design in figma
* Place those smaller images below the large image towards bottom left and increase it's size a little bit. it looks too small.
* You can look into the design for reference.
https://www.figma.com/file/figjD5PRyaAJ5ffTFVOMDN/Ecommerce-Site-Mern?type=design&node-id=38-383&mode=design&t=12x3SvJUpFQNYNcn-0

image

File: client/src/pages/product/product.jsx [hunk 1]
```diff
@@ -19,9 +22,25 @@ const ProductPage = () => {
...
+  //   ],
```
Comment (victorchrollo14) [inline client/src/pages/product/product.jsx:43]: Remove this when you make the next changes.

**Review by victorchrollo14 — CHANGES_REQUESTED** [review]

**Comment by victorchrollo14** [comment]
@Sky-De The slider works fine.Good work. 
Just make the above changes, then we can merge it up.

**Comment by Sky-De** [comment]
@victorchrollo14 ok i'll fix them, however i took a look at your figma design and there was no match with screenshot which you provided before



do you want design be match with this screen shot ?

**Comment by victorchrollo14** [comment]
> @victorchrollo14 ok i'll fix them, however i took a look at your figma design and there was no match with screenshot which you provided before
> do you want design be match with this screen shot ?

The screenshot is how the page looks with changes you made. Just make it as it looks like in the figma design

**Comment by Sky-De** [comment]
@victorchrollo14 it is done now, there is just a little change in arrow btns because with color of design (#000) there will hidden in images with white bg, also black for these arrows wont work because in other images with dark bg it will be hard to see
this screen shot is show case of issue :


asdasd


##  Gif of final result
as a result of bg problem which mentioned before, i had to change arrow styles a bit , then this is the final UI :

chrome-capture-2023-8-23-min (1)

If it works for you let me know to make commit of these changes
thanks

**Comment by victorchrollo14** [comment]
Ok That looks fine. Can you remove the extra white space to the right of the image and keep it like below. 


Product Page

**Comment by Sky-De** [comment]
@victorchrollo14 i reduced amount of white space and it is better now, if i want to make it exactly look like your design i have to change rightSection styles  which is totally other issue

## Gif

chrome-capture-2023-8-24-min


## consider
i made a commit and it is ready to use
last commit
thanks

**Comment by victorchrollo14** [comment]
> @victorchrollo14 i reduced amount of white space and it is better now, if i want to make it exactly look like your design i have to change rightSection styles which is totally other issue

You don't have to change the entire thing, I tried it, it's a simple fix.
max-w-[600px] add this to the div with className="left_section"
then remove lg:w-[70%] from Carousel  Component.
it would look like this when you make the changes.


image

**Review by victorchrollo14 — APPROVED** [review]

**Comment by victorchrollo14** [comment]
@Sky-De Good work. I was able to edit that directly, So I'm merging it up.

**Comment by Sky-De** [comment]
@victorchrollo14 Hi, may I ask you a favor? I'd appreciate it if you could take a look at my [Portfolio ](https://skyde-portfolio.netlify.app/)and other projects. Please let me know what you think about them, and if you like them, please give them a STAR. All public projects can be found in the PROJECTS section of my portfolio. Within that section, you can find live demo links as well as repository links related to each project. And if you like my project please mention other colleagues of these projects here to check my [Portfolio ](https://skyde-portfolio.netlify.app/), thanks thanks

**Comment by victorchrollo14** [comment]
your portfolio looks cool. what problem are you facing right now.
I too am a student just learning and building stuff.
just hit me up on twitter, we could have a chat.

**Comment by Sky-De** [comment]
> your portfolio looks cool. what problem are you facing right now. I too am a student just learning and building stuff. just hit me up on twitter, we could have a chat.

Thanks for looking at it, i don't have any problem with that, i just asked people who i contribute with to take a look and give my projects start, take a look at book sharing too, if you like give it star too and leave a feedback in that project itself
Thanks

**Comment by victorchrollo14** [comment]
> > your portfolio looks cool. what problem are you facing right now. I too am a student just learning and building stuff. just hit me up on twitter, we could have a chat.
> 
> Thanks for looking at it, i don't have any problem with that, i just asked people who i contribute with to take a look and give my projects start, take a look at book sharing too, if you like give it star too and leave a feedback in that project itself
> Thanks

Yeah sure.

**Pull request merged**